frontend/src/common/AnimatedList/AnimatedList.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {
+  Key,
+  RefObject,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
+import { List } from '@material-ui/core';
+
+/**
+ * GUI element of the {@link AnimatedList}.
+ */
+export type AnimatedListElement = JSX.Element & {
+  ref?: RefObject<HTMLElement>;
+};
+
+type Props = {
+  /**
+   * List-items; must have a key and a ref.
+   */
+  children: AnimatedListElement[];
+  className?: string;
+};
+
+/**
+ * Information about items in the AnimatedList-
+ */
+type ItemGuiInformation = {
+  /**
+   * Map with key = element key and value = bounding box.
+   */
+  boxes: Map<Key, DOMRect>;
+  /**
+   * Scroll position at the creation of objects of this type.
+   */
+  scrollY: number;
+};
+
+/**
+ * Returns an {@link ItemGuiInformation} object for the given list items
+ * @param children elements of the list
+ */
+function createItemGuiInformation(
+  children: AnimatedListElement[]
+): ItemGuiInformation {
+  const boundingBoxes = new Map<Key, DOMRect>();
+
+  React.Children.forEach(children, (child) => {
+    const domNode = child.ref?.current;
+    if (domNode && child.key != null) {
+      boundingBoxes.set(child.key, domNode.getBoundingClientRect());
+    }
+  });
+
+  return { boxes: boundingBoxes, scrollY: window.scrollY };
+}
+
+/**
+ * Outputs the previous state from a variable of useState.
+ * See https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state
+ */
+function usePrevious<T>(value: T) {
+  const prevValue = useRef<T>();
+
+  useEffect(() => {
+    prevValue.current = value;
+  }, [value]);
+
+  return prevValue.current;
+}
+
+/**
+ * Returns a list that animated the reordering of its children.
+ * @param children list-items
+ * @param className react standard className attribute
+ */
+function AnimatedList({ children, className }: Props): JSX.Element {
+  const [currBoundingBox, setCurrBoundingBox] = useState<ItemGuiInformation>({
+    boxes: new Map(),
+    scrollY: 0,
+  });
+  const prevBoundingBox = usePrevious(currBoundingBox);
+
+  useLayoutEffect(() => {
```

Comment:
Maybe write a short comment why you use `useLayoutEffect()` instead of `useEffect()` here.